### PR TITLE
feat(generator): new cmd to update googleapis-root

### LINKED ISSUE #271: Update the googleapis SHA and regenerate all the client libraries.

### PR DESCRIPTION
**Pull request by coryan**
The `update` command updates the `googleapis-root` SHA and then
refreshes all the generated code.

Fixes #271

### CODE REVIEW

**Comment by coryan** [comment]
The generator breakage is unrelated and will be fixed by #392

File: generator/internal/sidekick/update.go [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+	}
+	return refreshAll(rootConfig, cmdLine)
+}
+
+func updateRootConfig(rootConfig *Config) error {
+	gitHubApi, ok := rootConfig.Source["github-api"]
+	if !ok {
+		gitHubApi = defaultGitHubApi
+	}
+	gitHub, ok := rootConfig.Source["github"]
```
Comment (codyoss) [inline generator/internal/sidekick/update.go:53]: Note: I think this is fine for now. But in the future we may want to also honor an environment variable to source a local github token if we notice we are running into quota issues. There are some good libs out there that make all this pretty easy that I have used before.

File: generator/internal/sidekick/sidekick.go [hunk 2]
```diff
@@ -61,19 +61,14 @@ func runSidekick(cmdLine *CommandLine) error {
...
-		if err := refresh(config, cmdLine, cmdLine.Output); err != nil {
-			return err
-		}
+		return refresh(config, cmdLine, cmdLine.Output)
 	case "refresh-all", "refreshall":
-		if err := refreshAll(config, cmdLine); err != nil {
-			return err
-		}
+		return refreshAll(config, cmdLine)
+	case "update":
```
Comment (codyoss) [inline generator/internal/sidekick/sidekick.go:69]: Can we file a bug to add help text for these different commands?

**Review by codyoss — APPROVED** [review]

File: generator/internal/sidekick/sidekick.go (hunk 2 above)
Comment (coryan) [reply generator/internal/sidekick/sidekick.go:69]: Done (#393)

**Pull request merged**